datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -134,20 +134,20 @@ impl PhysicalExpr for ScalarFunctionExpr {
     }
 
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
```

Comment:
I think we could potentially increase the number of places arrays could be reused by changing this to take `batch` (owned) and threading that more fully through the physical exprs. 

Until there is some compelling usecase to do so, however, I don't think we should go mess with the APIs again